### PR TITLE
Fix for #41 - proxy support

### PR DESCRIPTION
**Pull request by genotrance**
Sourced getProxy() from Nimble

### CODE REVIEW

File: src/choosenim/download.nim [hunk 1]
```diff
@@ -138,8 +138,31 @@ when defined(curl):
...
+        getCurrentExceptionMsg(), Warning, HighPriority)
+
+  if url.len > 0:
+    var parsed = parseUri(url)
+    if parsed.scheme.len == 0 or parsed.hostname.len == 0:
+      parsed = parseUri("http://" & url)
+    let auth =
+      if parsed.username.len > 0: parsed.username & ":" & parsed.password
+      else: ""
+    return newProxy($parsed, auth)
```
Comment (dom96) [inline src/choosenim/download.nim]: I find this code odd. Doesn't ``httpclient`` provide this logic? If not, I think it should. But we can take care of that later.

Comment (genotrance) [reply src/choosenim/download.nim]: Nope, this isn't in httpclient.nim yet. I know you mentioned this separately as well but I added this to choosenim for the near term.

Comment (genotrance) [reply src/choosenim/download.nim]: Nope, this isn't in httpclient.nim yet. I know you mentioned this separately as well but I added this to choosenim for the near term.

**Comment by dom96** [comment]
This will do for now but I would like it in the stdlib.

**Pull request merged**